lib/kafka/producer_interceptors.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+module Kafka
+  # Holds a list of Kafka::ProducerInterceptor
+  # and wraps calls to a chain of custom producer interceptors.
+  class ProducerInterceptors
+    def initialize(interceptors:, logger:)
+      @interceptors = interceptors
+      @logger = TaggedLogger.new(logger)
+    end
+
+    # This method is called when the client produces a message, before it gets queued to be sent to the Kafka brokers.
+    # The message returned from the first ProducerInterceptor.on_send is passed to the second interceptor on_send, and so on in an
+    # interceptor chain. This method does not throw exceptions.
+    #
+    # @param pending_message [Kafka::PendingMessage] the produced message.
+    #
+    # @return [Kafka::PendingMessagePendingMessage] the intercepted message returned by the last interceptor.
+    def on_send(pending_message:)
+      intercepted_pending_message = pending_message
+      @interceptors.each do |interceptor|
+        begin
+          intercepted_pending_message = interceptor.on_send(pending_message: intercepted_pending_message)
+        rescue Exception => e
+          if intercepted_pending_message.nil?
```

Comment:
A method can't both return a value and throw an exception.
If we've caught a message, it means nothing was returned from `on_send`, and `intercepted_pending_message` will be equal to the same thing we passed in to the interceptor.

Is this just making it so we don't print the topic if someone intentionally tries to produce `nil`?

Reply:
it's because if it's `nil` I cannot add to the warn message info about the topic and partition